### ISSUE TITLE
Meson /  Ninja under MinGW64 mangled output, implicit Werror, etc

### ISSUE THREAD

**Issue by spl1inter** [msys2/MINGW-packages#5044]
Hi there, is there any special guidance for using the meson toolchain under Windows-MinGW64?

First the output is hard to read, probably due to python3 utf8 attempt to colorize its output incorrectly under windows, here attempting hexchat build:
```
meson mingw64 -Dprefix=/MSYS2/mingw64 -Dwith-libcanberra=false -Dwith-perl=false -Dwith-upd=false -Dwith-winamp=false -Dwith-appdata=false -Dwith-libnotify=false -Dwith-lua=false

←[1mThe Meson build system←[0m
Version: 0.49.0
Source dir: ←[1mD:\Download\MinGW-64-souorce\hexchat-2.14.2←[0m
Build dir: ←[1mD:\Download\MinGW-64-souorce\hexchat-2.14.2\mingw64←[0m
Build type: ←[1mnative build←[0m
Project name: ←[1mhexchat←[0m
Project version: ←[1m2.14.2←[0m
Native C compiler: ←[1mcc←[0m (gcc 8.2.0 "cc (Rev3, Built by MSYS2 project) 8.2.
0")
Build machine cpu family: ←[1mx86_64←[0m
Build machine cpu: ←[1mx86_64←[0m
Found pkg-config: ←[1mD:\MSYS2\mingw64\bin\pkg-config.EXE←[0m (0.29.2)
Dependency ←[1mgio-2.0←[0m found: ←[1;32mYES←[0m 2.58.2
...

```

Could it be this is due to python3 specific use of utf8 as sys.defaultencoding under windows?
If I change the console codepage using "chcp 65001", things get a _little_ better(?):
```[01m[K../src/common/text.c:203:26:[m[K [01;31m[Kerror: [m[Kformat '[01m
[K%i[m[K' expects argument of type '[01m[Kint[m[K', but argument 2 has ty
pe '[01m[Klong long int[m[K' [[01;31m[K-Werror=format=[m[K]
   buf = g_strdup_printf ([01;31m[K"T %"[m[K G_GINT64_FORMAT " ", [32m[K(g
int64)stamp[m[K);
                          [01;31m[K^~~~~[m[K                      [32m[K~~
~~~~~~~~~~~[m[K
In file included from [01m[KD:/MSYS2/mingw64/include/glib-2.0/glib/gtypes.h:32
[m[K,
                 from [01m[KD:/MSYS2/mingw64/include/glib-2.0/glib/galloca.h:3
2[m[K,
                 from [01m[KD:/MSYS2/mingw64/include/glib-2.0/glib.h:30[m[K,

                 from [01m[K../src/common/hexchat.h:22[m[K,
                 from [01m[K../src/common/text.c:35[m[K:
[01m[KD:/MSYS2/mingw64/include/glib-2.0/glibconfig.h:62:29:[m[K [01;36m[Kn
ote: [m[Kformat string is defined here
 #define G_GINT64_FORMAT "I64i"
cc1.exe: some warnings being treated as errors
[51/87] Compiling C object plugins/sys...rc_common_sysinfo_win32_backend.c.obj.
ninja: build stopped: subcommand failed.
```

Also notice apparent use of Werror CFLAG, etc. Any guidance on best practice fixing things like this under MinGW64? thanks.

**Comment by lazka** [comment]
Which terminal are you using?

**Comment by spl1inter** [comment]
Please reconsider. Implicit MSYS dependency on other msys applications makes sense, but the idea of MINGW is, well was, native WIN32 ports. Windows console can support colors as well. Otherwise no sense to make minw port, just use all msys.

**Comment by lazka** [comment]
I guess you are using a plain Windows terminal then?

**Comment by spl1inter** [comment]
Of course, as I would expect any of my port users to be using.

**Comment by lazka** [comment]
OK, I'll have a look.

**Comment by lazka** [comment]
Should be fixed by #5229
